pkg/monitoring/metrics/forklift-controller/migration_metrics.go

```diff
@@ -26,85 +29,91 @@ func RecordMigrationMetrics(c client.Client) {
 				continue
 			}
 
-			// Holding counter vars used to make gauge update "atomic"
-			var succeededRHV, succeededOCP, succeededOVA, succeededVsphere, succeededOpenstack float64
-			var failedRHV, failedOCP, failedOVA, failedVsphere, failedOpenstack float64
+			// Initialize or reset the counter map at the beginning of each iteration
+			counterMap := make(map[string]float64)
 
-			// for all migrations, count # in executing, succeeded, failed, canceled
 			for _, m := range migrations.Items {
-
 				plan := api.Plan{}
 				err := c.Get(context.TODO(), client.ObjectKey{Namespace: m.Spec.Plan.Namespace, Name: m.Spec.Plan.Name}, &plan)
 				if err != nil {
 					continue
 				}
 
-				provider := api.Provider{}
-				err = c.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Source.Namespace, Name: plan.Spec.Provider.Source.Name}, &provider)
+				sourceProvider := api.Provider{}
+				err = c.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Source.Namespace, Name: plan.Spec.Provider.Source.Name}, &sourceProvider)
+				if err != nil {
+					continue
+				}
+
+				destProvider := api.Provider{}
+				err = c.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Destination.Namespace, Name: plan.Spec.Provider.Destination.Name}, &destProvider)
 				if err != nil {
 					continue
 				}
 
+				isLocal := destProvider.Spec.URL == ""
+				isWarm := plan.Spec.Warm
+
+				var target, mode, key string
+				if isLocal {
+					target = Local
+				} else {
+					target = Remote
+				}
+				if isWarm {
+					mode = Warm
+				} else {
+					mode = Cold
+				}
+
+				provider := sourceProvider.Type().String()
+
 				if m.Status.HasCondition(Succeeded) {
-					switch provider.Type() {
-					case api.Ova:
-						succeededOVA++
-						continue
-					case api.OVirt:
-						succeededRHV++
-						continue
-					case api.VSphere:
-						succeededVsphere++
-						continue
-					case api.OpenShift:
-						succeededOCP++
-						continue
-					case api.OpenStack:
-						succeededOpenstack++
-						continue
+					key = fmt.Sprintf("%s|%s|%s|%s|%s", Succeeded, provider, mode, target, string(plan.UID))
+					counterMap[key]++
+
+					startTime := m.Status.Started.Time
+					endTime := m.Status.Completed.Time
+					duration := endTime.Sub(startTime).Seconds()
+
+					var totalDataTransferred float64
+					for _, vm := range m.Status.VMs {
+						for _, step := range vm.Pipeline {
+							if step.Name == "DiskTransferV2v" || step.Name == "DiskTransfer" {
+								for _, task := range step.Tasks {
+									totalDataTransferred += float64(task.Progress.Completed)
+								}
+							}
+						}
+					}
+
+					// Set the metrics for duration and data transferred and update the map for scaned migration
+					if _, exists := processedSucceededMigrations[string(m.UID)]; !exists {
```

Comment:
Since we don't want to add a new metric for each successful migration each iteration, once we process it once, we will save its ID to a map and ignore it in the next iterations, only for successful migrations.